content/blog/go-sdk-road-to-2/index.md

```diff
@@ -0,0 +1,67 @@
+---
+title: "Modern Cloud Infrastructure in Golang  - The Road to 2.0"
+authors: ["evan-boyle"]
+tags: ["go", "golang", "aws", "gcp", "azure"]
+date: "2020-02-25"
+meta_desc: "Pulumi + Go is a powerful combo for your cloud-native infrastructure."
+meta_image: "pulumigo.png"
+---
+
+Here at Pulumi, everyone on our engineering team is a Gopher. We build our open-source [pulumi/pulumi engine](https://github.com/pulumi/pulumi) and our SaaS with Go. Pulumi has supported Go in preview for more than a year, but we've recently invested in making Go support for Pulumi great as part of [Pulumi 2.0]({{< relref "/blog/pulumi-2-0-roadmap#better-language-support" >}}).
+
+## What is Pulumi?
+
+Pulumi lets you use real languages to express your application’s infrastructure needs, using a powerful technique called “infrastructure as code.” Using infrastructure as code, you declare desired infrastructure, and an engine provisions it for you so that it’s automated, easy to replicate, and robust enough for demanding production requirements. Pulumi improves this process by leveraging real languages and making modern cloud infrastructure patterns, such as containers and serverless programs, first-class and easy.
+
+Automatically create, update, or delete cloud resources using Pulumi’s infrastructure as code engine, removing manual point-and-clicking in the cloud provider consoles, UIs, and ad-hoc scripts.
+Use your favorite IDEs and tools, including Visual Studio Code, taking advantage of features like auto-completion, refactoring, and interactive documentation.
+Catch mistakes early on with standard compiler errors, and an infrastructure-specific policy engine for enforcing security, compliance, and best practices.
+Reuse any existing go packages, or distribute your own, whether that’s for infrastructure best practices, productivity, or just general programming patterns.
+Build scalable cloud applications using classic infrastructure cloud-native technologies. including serverless functions, and highly scalable databases such as AWS Aurora, into your core development experience, bringing them closer to your application code.
+
+Pulumi’s free open source SDK, which includes a CLI and assortment of libraries, enables these capabilities.
+
+Creating an AWS S3 Bucket is familiar and simple:
+
+```go
+package main
+
+import (
+    "github.com/pulumi/pulumi-aws/sdk/go/aws/s3"
+    "github.com/pulumi/pulumi/sdk/go/pulumi"
+)
+
+func main() {
+    pulumi.Run(func(ctx *pulumi.Context) error {
+        // Create an AWS resource (S3 Bucket)
+        bucket, err := s3.NewBucket(ctx, "my-bucket", nil)
+        if err != nil {
+            return err
+        }
+
+        // Export the name of the bucket
+        ctx.Export("bucketName", bucket.ID())
+        return nil
+    })
+}
+```
+
+## An Expressive and Idiomatic Type System
```

Comment:
One aspect of the Go SDK that is worth mentioning is that we designed it to give you a little more control, with the intent of being embeddable within larger systems. That's why, for instance, you get the `*pulumi.Context` object. It's true we stopped short of programmatically driving deployments (per lots of recent discussions), however we've actually seen that play out in practice with some Go-oriented customers.

Reply:
This seems like a topic that could benefit from a post of its own similar to the type system deep dive. Much of this is undocumented and doesn't have public prior art to reference.

The goal of this post is to drive people through to the getting started and examples pages. Do you think that adding a section on programmatic access would help to this end, or would that topic be better as a standalone with some richer content?